utils/s3/client.cc

```diff
@@ -84,34 +88,57 @@ future<> ignore_reply(const http::reply& rep, input_stream<char>&& in_) {
 client::client(std::string host, endpoint_config_ptr cfg, semaphore& mem, global_factory gf, private_tag, std::unique_ptr<aws::retry_strategy> rs)
         : _host(std::move(host))
         , _cfg(std::move(cfg))
+        , _creds_invalidation_timer([this] {
+            s3l.info("Credentials update attempt in background failed. Outdated credentials will be discarded, triggering synchronous re-obtainment"
+                     " attempts for future requests.");
+            _credentials = {};
+        })
+        , _creds_update_timer([this] {
+            std::ignore = [this]() -> future<> {
+                s3l.info("Update creds in the background");
+                _credentials = co_await _creds_provider_chain.get_aws_credentials();
+                _creds_invalidation_timer.rearm(_credentials.expires_at);
```

Comment:
What if we don't reach this place because previous co_await throws? Then the `authorize()` will be in charge of synchronously updating creds, but since it doesn't re-arm any timers, creds become "perpetual"?